extensions/Webhooks/Extension.pm

```diff
@@ -0,0 +1,221 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::Webhooks;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use base qw(Bugzilla::Extension);
+
+use Bugzilla::Component;
+use Bugzilla::Product;
+use Bugzilla::Constants;
+use Bugzilla::Error;
+use Bugzilla::Extension::Webhooks::Webhook;
+use Bugzilla::User;
+use Bugzilla::Logging;
+use Try::Tiny;
+
+#
+# installation
+#
+
+sub db_schema_abstract_schema {
+  my ($self, $args) = @_;
+  my $dbh = Bugzilla->dbh;
+  $args->{'schema'}->{'webhooks'} = {
+    FIELDS => [
+      id      => {TYPE => 'INTSERIAL', NOTNULL => 1, PRIMARYKEY => 1,},
+      user_id => {
+        TYPE       => 'INT3',
+        NOTNULL    => 1,
+        REFERENCES => {TABLE => 'profiles', COLUMN => 'userid', DELETE => 'CASCADE',}
+      },
+      name  => {TYPE    => 'VARCHAR(64)', NOTNULL => 1,},
+      url   => {TYPE    => 'VARCHAR(64)', NOTNULL => 1,},
+      event => {TYPE    => 'VARCHAR(64)', NOTNULL => 1,},
+      product_id   => {
+        TYPE       => 'INT2',
+        NOTNULL    => 1,
+        REFERENCES => {TABLE => 'products', COLUMN => 'id', DELETE => 'CASCADE',}
+      },
+      component_id => {
+        TYPE       => 'INT2',
+        NOTNULL    => 0,
+        REFERENCES => {TABLE => 'components', COLUMN => 'id', DELETE => 'CASCADE',}
+      }
+    ],
+  };
+}
+
```

Comment:
Let's add a UNIQUE INDEX for the user_id and name columns. That way a user cannot have more than one webhook with the same name.

```
INDEXES => [
      webhooks_userid_name_idx => {FIELDS => ['user_id', 'name'], TYPE => 'UNIQUE',},
    ],
```